Fable.sln does not restore

I'm not able to run `dotnet restore Fable.sln` on the latest main:

```
dotnet restore .\Fable.sln                                                                                                                                                                                                       ❮  6s 727ms    1
  Determining projects to restore...
C:\Users\nojaf\Projects\Fable\tests\React\Fable.Tests.React.fsproj : warning NU1608: Detected package version outside of dependency constraint: Fable.Jester 0.33.0 requires Fable.Core (>= 3.2.8 && < 4.0.0) but version Fable.Core 4.2.0 was resolved. [C
:\Users\nojaf\Projects\Fable\Fable.sln]
C:\Users\nojaf\Projects\Fable\tests\React\Fable.Tests.React.fsproj : warning NU1608: Detected package version outside of dependency constraint: Fable.ReactTestingLibrary 0.33.0 requires Fable.Core (>= 3.2.8 && < 4.0.0) but version Fable.Core 4.2.0 was
 resolved. [C:\Users\nojaf\Projects\Fable\Fable.sln]
C:\Users\nojaf\Projects\Fable\tests\Js\Adaptive\Fable.Tests.Adaptive.fsproj : error NU1202: Package Expecto 10.1.0 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package Expecto 10.1.0 supports: net6.0 (.NETCoreApp,Version=v6.0) [C
:\Users\nojaf\Projects\Fable\Fable.sln]
  Failed to restore C:\Users\nojaf\Projects\Fable\tests\Js\Adaptive\Fable.Tests.Adaptive.fsproj (in 194 ms).
  23 of 24 projects are up-to-date for restore.
```

I believe this goes undetected as the CI jobs don't restore any solution file.
I do find this rather annoying, would you accept a PR with a simple job that builds both solutions?

--- a/.editorconfig
+++ b/.editorconfig
@@ -11,6 +11,9 @@ indent_size = 4
 [*.ts]
 indent_size = 2
 
+[*.yml]
+indent_size = 2
+
 # Fantomas (see https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html)
 [*.{fs,fsx,fsi}]
 max_line_length = 80

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -28,6 +28,21 @@ jobs:
       - name: Check F# code
         run: dotnet fantomas . --check
 
+  # Verify that Fable.sln can be restored
+  # See: https://github.com/fable-compiler/Fable/issues/3687
+  restore-project:
+    name: Check that solution can be restored
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+
+      - name: Restore project
+        run: dotnet restore Fable.sln
+
   # Separate build job for JavaScript
   build-javascript:
     strategy:

